spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jms/artemis/ArtemisConnectionFactoryFactory.java

```diff
@@ -110,10 +110,11 @@ private ArtemisMode deduceMode() {
 			TransportConfiguration transportConfiguration = new TransportConfiguration(
 					InVMConnectorFactory.class.getName(),
 					this.properties.getEmbedded().generateTransportParameters());
-			ServerLocator serviceLocator = ActiveMQClient
-					.createServerLocatorWithoutHA(transportConfiguration);
-			return factoryClass.getConstructor(ServerLocator.class)
-					.newInstance(serviceLocator);
+			try (ServerLocator serviceLocator = ActiveMQClient
```

Comment:
This well close the `ServerLocator` before the connection factory can use it. It is eventually closed when the connection factory is closed so this change is unnecessary. I'll address this when merging.